src/epygram/fields/D3VectorField.py

```diff
@@ -65,6 +65,69 @@ def psikhi2uv(psi, khi):
     v.validity = psi.validity
     return make_vector_field(u, v)
 
+def uv2psikhi(u, v):
+    """
+    Compute streamfunction **psi** and velocity potential **khi**
+    as a D3VectorField (or subclass) from wind (on the grid).
+    """
+    # Check space
+    if not (u.spectral and v.spectral):
+        raise epygramError("Wind must be in spectral space to compute psi/khi.")
+
+    # Compute voriticity / divergence
+    uv = epygram.fields.make_vector_field(u, v)
+    vor, div = uv.compute_vordiv()
+    vor.gp2sp(u.spectral_geometry)
+    div.gp2sp(u.spectral_geometry)
+
+    if u.geometry['name'] == 'lambert':
```

Comment:
I think it's valid for any projected geometry, not only Lambert ? (i.e. Mercator and Stereopolar)
In which case you could adapt the if test to `u.geometry.projected_geometry` property ?